### DISCUSSION TITLE
Saving all buffers and stopping insert mode when focus is lost

### DISCUSSION THREAD

**Discussion by krishnakumarg1984** [neovim/neovim#24485] [Q&A]
I currently have the following autocmd in my `init.lua` inside a `vim.cmd[[ ]]` block:

``` vim
  augroup FocusLostAucmd
    autocmd!
    " autocmd FocusLost * stopinsert
  " Save all buffers when focus lost, ignoring warnings,
  " and return to normal mode.
  "
  " Ideally we'd have:
  "
  "   autocmd FocusLost * silent! wa stopinsert
  "
  " but stopinsert doesn't seem to work inside a FocusLost autocommand.
  " So we use a long-winded approach instead.
  "
  " https://groups.google.com/g/vim_use/c/K-xK6SKIDrM/m/uBbVEJCVjyQJ
    autocmd FocusLost * nested silent! wa
    autocmd FocusLost * if mode()[0] =~ 'i\|R' | call feedkeys("\<Esc>") | endif
  augroup END
```

This snippet is from a vim mailing list post from nearly 15 years ago. Is this still relevant in neovim, or have things been much simplified since in order to achieve this?

What would be the modern lua equivalent config lines to achieve this?

**Comment by nyngwang** [comment]
If you want autosave, this is what I have been using for almost a year, it doesn't fail once:

https://github.com/nyngwang/suave.lua/blob/3ac3d4096974ea0aefa9c713f47aaaff1088b30d/lua/suave/utils/autocmd.lua#L4_L20

**Comment by boorboor** [comment]
For a clean solution to save buffers on focus loss, consider using [Save.nvim](https://github.com/boorboor/save.nvim). It's a Lua plugin for Neovim that simplifies these tasks with easy configuration. Check it out!

**Comment by stCarolas** [comment]
```lua

vim.api.nvim_create_autocmd("FocusLost", {
	callback = function()
		vim.cmd("stopinsert")
	end,
})

```

**Reply by nyngwang** [reply]
You didn't save anything.
